### ISSUE TITLE
rule: ensure `vi.mock(<import>)` is pointing to a valid file

### ISSUE THREAD

**Issue by axel7083** [vitest-dev/eslint-plugin-vitest#665]
## Description

Dealing with large repository often lead to refactoring where we rename, update, delete some path. Most IDE are able to rename, and show where the import are. 

But when mocking a local import (https://vitest.dev/api/vi.html#vi-mock) we may have some tests failing, because we changed a filename, and finding where the problem is occurring might be difficult. 

A rule where we would ensure that the path mocked exists would be amazing (if feasible)

**Comment by veritem** [comment]
how would that rule help you specifically, check if a file exists?

**Comment by axel7083** [comment]
> how would that rule help you specifically, check if a file exists?

I can see several cases

- it would allow us to catch typo, if you misspelled a mock, it would tell you that the mock does not mock anything.
- when running the CI, the linter is usually run before the unit tests (at least for us), so it would be before the tests are ran stating: _the mock is not valid, you did a change that broke those tests_

I don't know exactly how internally it would work (never implemented any eslint before), checking that file exists would be a solution, telling you this does not exist etc.

**Comment by axel7083** [comment]
There is a workaround I found in the vitest documentation

Instead of writing

```ts
vi.mock('./utils/util', async () => {
  return {
    isMac: vi.fn(),
    isWindows: vi.fn(),
    getAssetsFolder: vi.fn(),
  };
});
```

You can write the following
vi.mock(import('./utils/util'), async () => {
  return {
    isMac: vi.fn(),
    isWindows: vi.fn(),
    getAssetsFolder: vi.fn(),
  };
});

Then if you have typescript properly configured, if the path does not exist you will get a nice error

```
error TS2307: Cannot find module './utils/uti455l' or its corresponding type declarations.

320 vi.mock(import('./utils/uti455l'), async (importActual) => {
```
